src/common.rs

```diff
@@ -1,52 +1,133 @@
-use near_cli_rs::common::JsonRpcClientExt;
-use near_jsonrpc_client::methods::{
-    validators::RpcValidatorRequest,
-    EXPERIMENTAL_genesis_config::{RpcGenesisConfigRequest, RpcGenesisConfigResponse},
-};
-use near_primitives::views::EpochValidatorInfo;
-
-fn validators_info(
-    epoch: near_primitives::types::EpochReference,
-    network_config: &near_cli_rs::config::NetworkConfig,
-) -> (RpcGenesisConfigResponse, EpochValidatorInfo) {
-    let client = network_config.json_rpc_client();
-    let genesis_config = client.blocking_call(RpcGenesisConfigRequest).unwrap();
-    let validator_info = client
-        .blocking_call(&RpcValidatorRequest {
-            epoch_reference: epoch,
+use color_eyre::eyre::ContextCompat;
+use num_rational::Rational32;
+
+use near_primitives::views::{CurrentEpochValidatorInfo, NextEpochValidatorInfo};
+
+pub fn find_seat_price(
+    validators: Vec<CurrentOrNextValidatorInfoOrProposalsTable>,
+    max_number_of_seats: u64,
+    minimum_stake_ratio: Rational32,
+    protocol_version: near_primitives::types::ProtocolVersion,
+) -> color_eyre::eyre::Result<near_cli_rs::common::NearBalance> {
+    if protocol_version < 49 {
+        return find_seat_price_for_protocol_before_49(validators, max_number_of_seats);
+    }
+    find_seat_price_for_protocol_after_49(
+        validators,
+        max_number_of_seats,
+        vec![*minimum_stake_ratio.numer(), *minimum_stake_ratio.denom()],
+    )
+}
+
+fn find_seat_price_for_protocol_before_49(
+    validators: Vec<CurrentOrNextValidatorInfoOrProposalsTable>,
+    num_seats: u64,
+) -> color_eyre::eyre::Result<near_cli_rs::common::NearBalance> {
+    let stakes = validators
+        .iter()
+        .map(|validator_info| match validator_info {
+            CurrentOrNextValidatorInfoOrProposalsTable::CurrentEpochValidatorInfo(current) => {
+                current.stake
+            }
+            CurrentOrNextValidatorInfoOrProposalsTable::NextEpochValidatorInfo(next) => next.stake,
+            CurrentOrNextValidatorInfoOrProposalsTable::ProposalsTable(proposals_table) => {
+                proposals_table.stake
+            }
+        })
+        .collect::<Vec<_>>();
```

Comment:
Given that `find_seat_price` only needs the vector of stakes, I would require it as a parameter instead of dealing with `CurrentOrNextValidatorInfoOrProposalsTable` (I believe that you don't need this enum at all)

Reply:
Done